### ISSUE TITLE
CreateOrUpdate on Entity with id which is CustomType doesn't work 

### ISSUE THREAD

**Issue by lstrzelecki** [j256/ormlite-core#44]
Hi

I have suspicion that on fields which are id and are custom type operation create or update doesn't work. Maybe I configured something wrong. Don't want to paste some code snippets. So I've created project which has this situation.

https://github.com/issue-example/ormlite.git on branch `issue/create_or_update_on_custom_type`

**Comment by j256** [comment]
Thanks for the branch and the test @lstrzelecki !!  I don't quite understand what makes this test special.  What do you mean "custom type operation"?  I certainly have tons of tests that do very similar things in the unit tests.  Here's one example:

https://github.com/j256/ormlite-core/blob/master/src/test/java/com/j256/ormlite/dao/BaseDaoImplTest.java#L70

**Comment by j256** [comment]
Just to be sure, I loaded this test into my code and it seems to run fine.

**Comment by lstrzelecki** [comment]
Hmmm

```
./gradlew test
:test
com.github.issue_example.ormlite.CreateOrUpdateTest > shouldStoreEntityWithCustomTypeIdAndLoadFromDbOnCreateOrUpdate FAILED
    org.h2.jdbc.JdbcSQLException at CreateOrUpdateTest.java:97
        Caused by: java.io.NotSerializableException at CreateOrUpdateTest.java:97

4 tests completed, 1 failed
:test FAILED
```
In BaseDaoImplTest which you shown there is Entity with integer, long or UUID primary key. In my case I have custom type which is in DB like VARCHAR. But when I try to call createOrUpdate method.
I have exception.
```java
org.h2.jdbc.JdbcSQLException: Serialization failed, cause: "java.io.NotSerializableException: com.github.issue_example.ormlite.CreateOrUpdateTest$EntityId"
```

So my custom Type is VARCHAR in DB. So during calling method ```com.j256.ormlite.dao.BaseDaoImpl#createOrUpdate```
and checking ```com.j256.ormlite.dao.BaseDaoImpl#idExists``` given ID object is treated as regular object and try to Serialze object into blob column. But its string according to custom type which is 
```java
com.github.issue_example.ormlite.CreateOrUpdateTest.EntityIdType extends com.j256.ormlite.field.types.StringType
```
Type is registered by ```DataPersisterManager.registerDataPersisters(EntityIdType.getInstance());``` and interpreted as well because I have Table with VARCHAR type column

**Comment by j256** [comment]
I don't see CreateOrUpdateTest in the branch.  What am I missing?

**Comment by lstrzelecki** [comment]
https://github.com/issue-example/ormlite/blob/issue/create_or_update_on_custom_type/src/test/java/com/github/issue_example/ormlite/CreateOrUpdateTest.java

But I already created testcase for that scenario. Please forgive me if I changed too much current testing code style.

**Comment by lstrzelecki** [comment]
Sorry for disturbing you.
1. Did you find Test with custom type?
2. Is it valid test case?
3. If it's an valid Issue is there any chance to solve that?

**Comment by j256** [comment]
Ok got it.  I've refactored your CreateOrUpdateTest.java here and fixed the bug.  Thanks for your patience.  Here's the code: https://github.com/j256/ormlite-core/commit/76be902f16bf4fe8497c105e04b31adaa0ee5e7e  Here's the bug report: https://sourceforge.net/p/ormlite/bugs/193/

**Comment by lstrzelecki** [comment]
Thank you very much for providing fix.
Wait time for fix was extremely  short. 

Thank you for this very good library.

PS
Fixed version should be more than 4.49. If fix is provided in core module there exists 4.49 since Dec 17, 2013

**Comment by j256** [comment]
4.48 was in 2013.  4.49 is RSN.
